### PR TITLE
Allow user to specify a CF for SST files generated by SstFileWriter

### PR DESCRIPTION
**Pull request by IslamAbdelRahman**
Allow user to explicitly specify that the generated file by SstFileWriter will be ingested in a specific CF.
This allow us to persist the CF id in the generated file

### CODE REVIEW

File: include/rocksdb/sst_file_writer.h [hunk 1]
```diff
@@ -43,8 +43,12 @@ struct ExternalSstFileInfo {
 // All keys in files generated by SstFileWriter will have sequence number = 0
 class SstFileWriter {
  public:
+  // User can pass `column_family` to specify that the the generated file will
+  // be ingested into this column_family, note that passing nullptr means that
+  // the column_family is unknown.
   SstFileWriter(const EnvOptions& env_options, const Options& options,
-                const Comparator* user_comparator);
+                const Comparator* user_comparator,
+                ColumnFamilyHandle* column_family = nullptr);
```
Comment (siying) [inline include/rocksdb/sst_file_writer.h]: I think we should allow them to simply pass in column family ID. For some systems, this file may be generated from some other places, e.g. data warehouse. They don't have access to the handle but they may know the column family ID.

Comment (siying) [inline include/rocksdb/sst_file_writer.h]: Another reason for passing in column family ID is that, if a user knows they have only one CF, they can always pass default column family ID, without having a DB opened.

File: table/sst_file_writer.cc [hunk 2]
```diff
@@ -36,12 +37,14 @@ struct SstFileWriter::Rep {
   ExternalSstFileInfo file_info;
   std::string column_family_name;
   InternalKey ikey;
+  ColumnFamilyHandle* cfh;
```
Comment (jkedgar) [inline table/sst_file_writer.cc]: Right now we never modify the data in the ColumnFamilyHandle, can this be a `const` pointer?

**Comment by IslamAbdelRahman** [comment]
Address @jkedgar comment and add a unit test

**Comment by IslamAbdelRahman** [comment]
@siying, I think it's better to let user pass cfh instead of cf_id .. because now we will leave TableProperties::column_family_name empty because we dont know it

**Comment by siying** [comment]
@IslamAbdelRahman OK.

**Pull request closed without merging**